lib/captain_hook.rb

```diff
@@ -14,6 +14,11 @@ def self.included(base)
     end
   end
 
+  class << self
+    attr_accessor :cache_key
+  end
+  self.cache_key = 0
```

Comment:
This certainly makes the trick. But I'm wondering if there's a more elegant way to approach this memoization...

BTW, I'm testing this version here: https://github.com/factorialco/factorial/pull/57002